packages/core/src/i18n/detect-language.ts

```diff
@@ -0,0 +1,53 @@
+import { IncomingHttpHeaders } from 'http';
+
+import { Optional } from '@logto/essentials';
+import { ParameterizedContext } from 'koa';
+import { IRouterParamContext } from 'koa-router';
+
+/**
+ * Resolve language and its q value from string.
+ * @param languageString The language string in header, e.g. 'en-GB;q=0.8', 'zh-CN'
+ * @returns `[language, q]`, e.g. `['en-GB', 0.8]`; `undefined` if no language is detected.
+ */
+const resolveLanguage = (languageString: string): Optional<[string, number]> => {
+  // Edited from https://github.com/lxzxl/koa-i18next-detector/blob/master/src/lookups/header.js
+  const [language, ...rest] = languageString.split(';');
+
+  if (!language) {
+    return;
+  }
+
+  for (const item of rest) {
+    const [key, value] = item.split('=');
+    if (key === 'q' && !Number.isNaN(Number(value))) {
```

Comment:
Maybe `Number.isNaN(Number(value)) => isNaN(value)` ?   isNaN should automatically coerce argument to numeric type.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#confusing_special-case_behavior

Reply:
![image](https://user-images.githubusercontent.com/14722250/131650856-653a95ba-3ccf-40b3-9c13-cb3a09305a4b.png)

![image](https://user-images.githubusercontent.com/14722250/131651043-f0a83e56-052b-4be6-a1bc-77f573734464.png)


sry, the global `isNaN()` only accepts `number`, and the global version is banned in our eslint ruleset. didn't notice `Number.isNaN()` accepts `unknown` value. updated.